### PR TITLE
Make `ExecuteBlock::execute_block` return the final block header

### PR DESCRIPTION
**Pull request by bkchr**
This pr changes the `ExecuteBlock` trait to return the final header that
results from executing the given block.

polkadot companion: https://github.com/paritytech/polkadot/pull/2548

### CODE REVIEW

**Review by thiolliere — APPROVED** [review]

**Review by shawntabrizi — APPROVED** [review]

**Comment by bkchr** [comment]
bot merge force

File: frame/support/src/traits.rs [hunk 1]
```diff
@@ -2228,6 +2228,23 @@ pub trait GetPalletVersion {
...
+	/// Execute the given `block`.
+	///
+	/// This will execute all extrinsics in the block and check that the resulting header is correct.
+	///
+	/// Returns the result header.
+	///
+	/// # Panic
+	///
+	/// Panics when an extrinsics panics or the resulting header doesn't match the expected header.
+	fn execute_block(block: Block) -> Block::Header;
```
Comment (kianenigma) [inline frame/support/src/traits.rs:2245]: doesn't the block already include a header, where it is equal to the one being returned?

Comment (bkchr) [reply frame/support/src/traits.rs:2245]: Yeah you are right. I also thought about this yesterday and than I thought that I had found something that requires this, but after thinking about this again, I do not require to return the header here. :(

Comment (kianenigma) [reply frame/support/src/traits.rs:2245]: will we revert?

**Pull request merged**